platform-tests/src/test/java/org/junit/platform/commons/util/ClassNameFilterUtilTests.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExecutionCondition;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.platform.commons.util.classes.AExecutionConditionClass;
+import org.junit.platform.commons.util.classes.ATestExecutionListenerClass;
+import org.junit.platform.commons.util.classes.AVanillaEmpty;
+import org.junit.platform.commons.util.classes.BExecutionConditionClass;
+import org.junit.platform.commons.util.classes.BTestExecutionListenerClass;
+import org.junit.platform.commons.util.classes.BVanillaEmpty;
+import org.junit.platform.launcher.TestExecutionListener;
+
+/**
+ * Unit tests for {@link ClassNameFilterUtil}.
+ *
+ * @since 5.7
+ */
+@SuppressWarnings({ "unchecked", "unused" })
+@TestInstance(Lifecycle.PER_CLASS)
+class ClassNameFilterUtilTests {
+
+	//@formatter:off
+	Stream<Arguments> alwaysEnabledConditions() {
+		return Stream.of(
+				arguments("org.junit.jupiter.*"),
+				arguments("org.junit.platform.*.NonExistentClass"),
+				arguments("*.NonExistentClass*"),
+				arguments("*NonExistentClass*"),
+				arguments("AExecutionConditionClass, BExecutionConditionClass"));
+	}
+
+	@MethodSource
```

Comment:
I'd rather use `@ValueSource(strings = {})` for these tests.

Reply:
Not sure why I didn't think of that.